src/pygama/dsp/processors/linear_slope_fit.py

```diff
@@ -75,3 +75,59 @@ def linear_slope_fit(
 
     slope[0] = (isum * sum_xy - sum_x * sum_y) / (isum * sum_x2 - sum_x * sum_x)
     intercept[0] = (sum_y - sum_x * slope[0]) / isum
+
+
+@guvectorize(
+    [
+        "void(float32[:], float32[:], float32[:], float32[:], float32[:])",
+        "void(float64[:], float64[:], float64[:], float64[:], float64[:])",
+    ],
+    "(n),(),()->(),()",
+    **nb_kwargs,
+)
+def linear_slope_diff(
+    w_in: np.ndarray, slope: float, intercept: float, mean: float, rms: float
+) -> None:
+    """
+    Calculate the mean and standard deviation of the waveform using
+    Welford's method as well as the slope an intercept of the waveform
+    using linear regression.
+    Parameters
+    ----------
+    w_in
+        the input waveform.
+    mean
+        the mean of the waveform.
+    stdev
+        the standard deviation of the waveform.
+    slope
+        the slope of the linear fit.
+    intercept
+        the intercept of the linear fit.
+    JSON Configuration Example
+    --------------------------
+    .. code-block :: json
+        "bl_mean, bl_std, bl_slope, bl_intercept": {
+            "function": "linear_slope_fit",
+            "module": "pygama.dsp.processors",
+            "args": ["wf_blsub[0:1650]", "bl_mean", "bl_std", "bl_slope", "bl_intercept"],
+            "unit": ["ADC", "ADC", "ADC", "ADC"],
+        }
```

Comment:
Won't render without spaces:
```suggestion
    Calculate the mean, standard deviation, slope and y-intercept of waveform.
    
    Uses Welford's method and linear regression.

    Parameters
    ----------
    w_in
        the input waveform.
    mean
        the mean of the waveform.
    stdev
        the standard deviation of the waveform.
    slope
        the slope of the linear fit.
    intercept
        the intercept of the linear fit.

    JSON Configuration Example
    --------------------------

    .. code-block :: json
        "bl_mean, bl_std, bl_slope, bl_intercept": {
            "function": "linear_slope_fit",
            "module": "pygama.dsp.processors",
            "args": ["wf_blsub[0:1650]", "bl_mean", "bl_std", "bl_slope", "bl_intercept"],
            "unit": ["ADC", "ADC", "ADC", "ADC"],
        }
```